src/Overlay.js

```diff
@@ -42,56 +35,63 @@ class Overlay extends React.Component {
       , animation: Transition
       , ...props } = this.props;
 
-    let child = null;
-
-    if ( Transition === true ){
+    if (Transition === true) {
       Transition = Fade;
     }
 
-    if (props.show || (Transition && this.state.exiting && !this.state.exited)) {
+    // Don't un-render the overlay while it's transitioning out.
+    const mountOverlay = props.show || (Transition && !this.state.exited);
```

Comment:
What was the rational for removing `this.state.exiting`? I originally had the code like this, but ran into a bug...that I can't remember, that this additional check fixed